reassembly_queue.go

```diff
@@ -216,15 +216,15 @@ func (r *reassemblyQueue) findCompleteUnorderedChunkSet() *chunkSet {
 	}
 
 	// Extract the range of chunks
-	chunks := r.unorderedChunks[startIdx : startIdx+nChunks]
+	var chunks []*chunkPayloadData
+	chunks = append(chunks, r.unorderedChunks[startIdx:startIdx+nChunks]...)
+
 	r.unorderedChunks = append(
 		r.unorderedChunks[:startIdx],
 		r.unorderedChunks[startIdx+nChunks:]...)
 
 	chunkSet := newChunkSet(0, chunks[0].payloadType)
-	for _, c := range chunks {
-		chunkSet.push(c)
-	}
+	chunkSet.chunks = chunks
```

Comment:
These lines were the root cause. `chunks` which was sliced out from `r.unorderedChunks`, was corrupted when `r.unorderedChunks` was altered. (see [this example](https://play.golang.org/p/v1qJg4DOR0K))